src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1637,27 +1637,13 @@ private void MakeSeparatorList(ReadOnlySpan<char> separators, ref ValueListBuild
             }
 
             // Special-case the common cases of 1, 2, and 3 separators, with manual comparisons against each separator.
-            else if (separators.Length <= 3)
+            else if (separators.Length <= 3u)
```

Comment:
does it affect codegen?

Reply:
Yes, it got rid of redundant range checks for `separators`, doing `(uint)separators.Length <= (uint)3` is one movsxd less, but I personally thought this was cleaner. However, I can see it being too obscure with it's intent.